tests/api/documents/test_documents_cover_api.py

```
@@ -0,0 +1,166 @@
+
+        # Verify failure response
+        assert data["success"] is False
+        assert data["isbn"] == isbn
+        assert "image" not in data
+        assert "provider" not in data
+
+        # Verify get_thumbnail_url was called
+        mock_get_thumbnail.assert_called_once_with(isbn, cached=True)
+
+
+def test_cover_endpoint_various_isbn_formats(client, app):
```

Ruff (0.15.1)

[warning] Unused function argument: `app`

(ARG001)